pkg/controller/job/job_controller.go

```diff
@@ -475,29 +474,40 @@ func (jm *JobController) patchJobResource(j *batch.Job, pods []*v1.Pod) error {
 
 		resourceUpdates := mergeResourceChanges(pod, cMap)
 		if len(resourceUpdates) > 0 {
+			if requestVer, ok := pod.ObjectMeta.Annotations[schedulerapi.AnnotationResizeResourcesRequestVer]; ok && requestVer == j.ResourceVersion {
+				// This is not a new request, check if earlier request failed.
+				// TODO: Implement better scheme - exponential backoff/pod-leaving event.. For now update based retry - yuck.
+				skip := true
+				for i, podCondition := range pod.Status.Conditions {
+					if podCondition.Type == v1.PodResourcesResizeStatus && podCondition.Status == v1.ConditionFalse {
+						if time.Since(podCondition.LastProbeTime.Time).Seconds() > 30 {
```

Comment:
30 should be in a constant variable with name explaining it's purpose

Reply:
yes. this can be cleaned up when we implement a good retry mechanism that is based on pod updates where pods leave the cluster (or node in case of the 'InPlaceOnly' policy. i've added a tracking item.